diff -urN user_import_orig/supported/field.inc user_import/supported/field.inc
--- user_import_orig/supported/field.inc	2013-08-12 00:29:11.000000000 +0200
+++ user_import/supported/field.inc	2013-12-04 11:25:28.000000000 +0100
@@ -181,9 +181,55 @@
 //      'save' => 'field_user_import_default_field_processor',
 //    );
 
+    if (module_exists('fedict_beidmellon')) {
+      $supported['fedict_fedid'] = array(
+        'save'     => 'field_user_import_fedict_fedid_processor',
+      );
+    }
+
     // hook to add functions to this array.
 
   }
 
   return $supported;
 }
+
+/**
+ * Copied from field_user_import_default_field_processor.
+ */
+function field_user_import_fedict_fedid_processor($user_fields, $field_name, $values) {
+  try {
+    $field = $user_fields->$field_name;
+
+    // VALIDATE
+    $fedid = $values[0];
+    // Check for non allowed characters (ie. not numbers, spaces, dots or dashes)
+    $allowed_characters_pattern = "/^[0-9A-z]+$/";
+
+    if (preg_match($allowed_characters_pattern, $fedid) != 1) {
+      throw new Exception(t("Non allowed character in FedID."));
+    }
+    else {
+      // Length must be 40 characters.
+      if (strlen($fedid) != 40) {
+        throw new Exception(t("Length of the FedID is not 40 characters."));
+      }
+    }
+    $query = new EntityFieldQuery;
+    $query->entityCondition('entity_type', 'user')
+    ->fieldCondition($field_name, 'fedict_fedid', $fedid)
+    ->propertyCondition('uid', $user_fields->uid, '!=')
+    ->count();
+    $count = $query->execute();
+    if ($count > 0) {
+      throw new Exception(t('The submitted FedID is already being used by another account. It is not possible to have more than one account with the same FedID:'));
+    }
+    // SAVE
+    $field[LANGUAGE_NONE][0]['fedict_fedid'] = _fedict_fedid_hash($fedid);
+  }
+  catch (Exception $e) {
+    drupal_set_message(t('The user %uid has been saved but the FedID for this user has not been saved: ', array('%uid' => $user_fields->uid)) .  $e->getMessage(), 'warning');
+  }
+
+  return $field;
+}
diff -urN user_import_orig/supported/user.inc user_import/supported/user.inc
--- user_import_orig/supported/user.inc	2013-08-12 00:29:11.000000000 +0200
+++ user_import/supported/user.inc	2013-12-04 12:03:15.000000000 +0100
@@ -147,7 +147,7 @@
   }
   else {
 
-    $account_add['timezone'] = '-18000';
+    $account_add['timezone'] = 'Europe/Brussels';
     $account_add['status']   = 1;
     $account_add['init']     = $fields['user']['email'][0];
     $account_add['pass']     = (empty($fields['user']['password'][0])) ? user_password() : $fields['user']['password'][0];
